clang/test/SemaSYCL/spec_const_accesors.cpp

```diff
@@ -0,0 +1,46 @@
+// RUN: %clangxx -fsycl-device-only %s -o %t.bc
+//
+//==----------- spec_const_accessors.cpp -----------------------------------==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+// The test checks that Clang doesn't crash if multiple accessors are used
+// together with specialization constants in single kernel.
+
+#include <CL/sycl.hpp>
+
+using namespace sycl;
+
+class MyInt32Const;
+class MyKernel;
+
+int32_t val = 10;
+
+int main() {
+  cl::sycl::queue queue;
+  cl::sycl::program program(queue.get_context());
+
+  cl::sycl::experimental::spec_constant<int32_t, MyInt32Const> i32 =
+      program.set_spec_constant<MyInt32Const>(val);
+
+  program.build_with_kernel_type<MyKernel>();
```

Comment:
Yep, totally agree with Alexey, you need to reduce the test.
It seems that to test your fix you only need to capture two accessors and spec constant in a kernel, that is all, no need to create queue, programs, etc. Take a look into neighbor tests in SemaSYCL directory.

Reply:
Done